packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -418,26 +417,37 @@ export function restoreControlledInputState(element: Element, props: Object) {
   }
 }
 
-// In Chrome, assigning defaultValue to certain input types triggers input validation.
-// For number inputs, the display value loses trailing decimal points. For email inputs,
-// Chrome raises "The specified value <x> is not a valid email address".
-//
-// Here we check to see if the defaultValue has actually changed, avoiding these problems
-// when the user is inputting text
-//
-// https://github.com/facebook/react/issues/7253
+// Here we check to see if the defaultValue has actually changed and update it
 export function setDefaultValue(
   node: HTMLInputElement,
   type: ?string,
-  value: ToStringValue,
+  value: mixed,
+  force: ?boolean,
 ) {
   if (
-    // Focused number inputs synchronize on blur. See ChangeEventPlugin.js
-    type !== 'number' ||
-    getActiveElement(node.ownerDocument) !== node
+    type === 'button' ||
+    type === 'file' ||
+    type === 'hidden' ||
+    type === 'image' ||
+    type === 'reset' ||
+    type === 'submit'
```

Comment:
these are the types where setting .value sets the value attribute, yes?

should radio and checked be in this list as well? https://html.spec.whatwg.org/multipage/input.html#dom-input-value-default-on

Reply:
These are fields for which either dynamic value replacement is not expected, or there is no value at all.

For checkbox and radio, separate conditions with different processing are needed (_f.e. changing defaultChecked instead of defaultValue, and other specifics_). I mentioned above that if the current solution will be fine, I'll try to add processing for these types as well

Reply:
Removed this condition so that only changes for email and number fields remain in PR